File: Polimorfismo/ejercicio 2/Matriz.java
package polimorfismo;
import java.util.Arrays;
public class Matriz {
    private float[][] matriz = new float[10][10];

    public Matriz() {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                matriz[i][j] = (i == j) ? 1 : 0;
            }
        }
    }

    public Matriz(float[][] matriz) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                this.matriz[i][j] = matriz[i][j];
            }
        }
    }

    public Matriz sumar(Matriz m) {
        float[][] resultado = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                resultado[i][j] = this.matriz[i][j] + m.matriz[i][j];
            }
        }
        return new Matriz(resultado);
    }

    public Matriz restar(Matriz m) {
        float[][] resultado = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                resultado[i][j] = this.matriz[i][j] - m.matriz[i][j];
            }
        }
        return new Matriz(resultado);
    }

    public boolean igual(Matriz m) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (this.matriz[i][j] != m.matriz[i][j]) {
                    return false;
                }
            }
        }
        return true;
    }
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < 10; i++) {
            sb.append(Arrays.toString(matriz[i])).append("\n");
        }
        return sb.toString();
    }
}

File: Polimorfismo/ejercicio 2/Matriz.py
from multimethod import multimethod

class Matriz:
    
    @multimethod
    def __init__(self):

        self.matriz = [[1 if i == j else 0 for j in range(10)] for i in range(10)]

    @multimethod
    def __init__(self, matriz: list):
        self.matriz = [row[:] for row in matriz]

    @multimethod
    def sumar(self, otra: 'Matriz'):
        resultado = [[self.matriz[i][j] + otra.matriz[i][j] for j in range(10)] for i in range(10)]
        return Matriz(resultado)

    @multimethod
    def restar(self, otra: 'Matriz'):
        resultado = [[self.matriz[i][j] - otra.matriz[i][j] for j in range(10)] for i in range(10)]
        return Matriz(resultado)

    @multimethod
    def igual(self, otra: 'Matriz'):
        for i in range(10):
            for j in range(10):
                if self.matriz[i][j] != otra.matriz[i][j]:
                    return False
        return True

    def __str__(self):
        return "\n".join([str(row) for row in self.matriz])

m1 = Matriz()

valores = [[i + j for j in range(10)] for i in range(10)]
m2 = Matriz(valores)

suma = m1.sumar(m2)

resta = m2.restar(m1)

iguales = m1.igual(m2)

print("Matriz 1 (Identidad):")
print(m1)
print("\nMatriz 2:")
print(m2)
print("\nSuma:")
print(suma)
print("\nResta:")
print(resta)
print("\n¿Matriz 1 es igual a Matriz 2?", iguales)